src/main/scala/akka/persistence/cassandra/query/SequenceNumbers.scala

```diff
@@ -0,0 +1,43 @@
+package akka.persistence.cassandra.query
+
+private[query] object SequenceNumbers {
+  val empty = SequenceNumbers(Map.empty, Map.empty)
+
+  sealed trait Answer
+  case object Yes extends Answer
+  case object Before extends Answer
+  case object After extends Answer
+  case object PossiblyFirst extends Answer
+}
+
+private[query] final case class SequenceNumbers(
```

Comment:
This would be a nice state monad :) The current implementation is nice too though.